lib/fileAttachments.ts

```diff
@@ -0,0 +1,48 @@
+import { logMessage } from "@lib/logger";
+
+/**
+ * Given a form submission JSON string, returns an array of file attachments objects that
+ * includes the file name, S3 object URL and antivirus scan results.
+ * @param submissionID - The form submission ID
+ * @param formSubmission - The form submission to get the file attachments from
+ * @returns {{ fileName: string }[]} - Array of file attachment objects
+ */
+export function getFileAttachments(
+  submissionID: string,
+  formSubmission: string
+): { fileName: string }[] {
+  let attachments: { fileName: string }[] = [];
+  const attachmentPrefix = "form_attachments/";
+  const attachmentRegex =
+    /^form_attachments\/\d{4}-\d{2}-\d{2}\/\w{8}-\w{4}-\w{4}-\w{4}-\w{12}\/.+$/;
+
+  // Check file attachments exist by checking the form submission for the
+  // file attachment prefix that's part of the uploaded file's S3 object key
+  if (formSubmission.indexOf(attachmentPrefix) > -1) {
+    logMessage.info(`${submissionID}: Parsing file attachments from form submission`);
+
+    // Be more explicit in the attachment check to avoid false positives
+    const jsonObj = JSON.parse(formSubmission);
+    attachments = Object.values<string>(jsonObj)
+      .filter((value) => attachmentRegex.test(value))
+      .map((value) => ({ fileName: getFileNameFromPath(value) }));
+  }
+
+  return attachments;
+}
+
+/**
+ * Given a file path returns the file name.
+ * @param filePath - The file path to get the file name from
+ * @returns {string} - The file name
+ */
+export function getFileNameFromPath(filePath: string): string {
+  const lastSlash = filePath.lastIndexOf("/");
+  let fileName = filePath;
+
+  // Is there at least one slash that isn't the last character in the string?
+  if (lastSlash !== -1 && lastSlash !== filePath.length - 1) {
+    fileName = filePath.substring(lastSlash + 1);
+  }
+  return fileName;
```

Comment:
I'm pretty sure this entire function could be simplified to:

`return filePath.substring(filePath.lastIndexOf("/") + 1);`

Reply:
Ah nice, that's way cleaner.  It means this case won't work like I've got it in the unit test, but this isn't really a valid scenario either since S3 won't allow an empty filename or a filename with any slashes:
```javascript
getFileNameFromPath("form_attachments/") === "form_attachments/"
```
I'll get the code and tests updated 👍🏻